src/TextArea.tsx

```diff
@@ -0,0 +1,65 @@
+import clsx from 'clsx'
+import {
+	forwardRef,
+	InputHTMLAttributes,
+	useContext,
+	useRef,
+	useState,
+} from 'react'
+import { ThemeContext } from './context/ThemeContext'
+import { mergeRefs } from './utils/warning'
+import Label from './Label'
+
+type TextareaProps = InputHTMLAttributes<HTMLTextAreaElement> & {
+	invalid?: boolean
+	className?: string
+	disabled?: boolean
+	label?: string
+	name?: string
+}
+
+const TextArea = forwardRef<HTMLTextAreaElement, TextareaProps>(
+	(props, ref) => {
+		const { className, invalid, disabled, label, name, ...other } = props
+		const { textarea } = useContext(ThemeContext)
+		const textareaRef = useRef<HTMLTextAreaElement>(null)
+		const [value, setValue] = useState(props.value || '')
+
+		const baseStyles = textarea.base
+		const invalidStyles = textarea.invalid
+		const disabledStyles = textarea.disabled
+
+		const hasValidation = invalid !== undefined
+		const validationStyle = hasValidation
+			? invalid
+				? invalidStyles
+				: null
+			: null
+
+		const cls = clsx(
+			baseStyles,
+			// don't apply disabledStyle if has valid
+			!hasValidation && disabled && disabledStyles,
+			validationStyle,
+			className,
+		)
+		const textareaName = name ? name : 'textarea'
+
+		return (
+			<div className="relative">
+				{label && <Label htmlFor={textareaName}>{label}</Label>}
+				<textarea
+					name={textareaName}
+					className={cls}
+					ref={mergeRefs([ref, textareaRef])}
+					value={value}
+					disabled={disabled}
+					onChange={e => setValue(e.target.value)}
+					{...other}
+				></textarea>
```

Comment:
`textarea` should also have `{children}` so that user can pass a default value if required.
`children` should be optional